centreon/src/Core/Security/Authentication/Application/UseCase/Login/LoginRequest.php

```diff
@@ -87,4 +91,13 @@ public static function createForSSO(string $clientIp): self
     {
         return new self(Provider::WEB_SSO, $clientIp);
     }
+
+    /**
+     * @param string $clientIp
+     * @return LoginRequest
+     */
+    public static function createForSAML(string $clientIp): self
+    {
+        return new self(providerName: Provider::SAML, clientIp: $clientIp);
```

Comment:
we should avoid named properties ATM
```suggestion
        return new self(Provider::SAML, $clientIp);
```

Reply:
At the beginning of my development I had many properties, it's an oversight